src/components/favorites.tsx

```diff
@@ -1,43 +1,47 @@
           <section className="favorites">
             <h1 className="favorites__title">Saved listing</h1>
             <ul className="favorites__list">
-              <li className="favorites__locations-items">
-                <div className="favorites__locations locations locations--current">
-                  <div className="locations__item">
-                    <a className="locations__item-link" href="#">
-                      <span>Amsterdam</span>
-                    </a>
-                  </div>
-                </div>
-                <div className="favorites__places">
-                  { favoritesCards.slice(0, 2).map((card) =>
-                    <Card key={''} typeCard={card.typeCard} isPremium={card.isPremium} image={card.image} price={card.price} rating={card.rating * 20} text={card.text} type={card.type}/>
-                  )}
-                </div>
-              </li>
-
-              <li className="favorites__locations-items">
-                <div className="favorites__locations locations locations--current">
-                  <div className="locations__item">
-                    <a className="locations__item-link" href="#">
-                      <span>Cologne</span>
-                    </a>
-                  </div>
-                </div>
-                <div className="favorites__places">
-                  { favoritesCards.slice(2).map((card) =>
-                    <Card key={''} typeCard={card.typeCard} isPremium={card.isPremium} image={card.image} price={card.price} rating={card.rating * 20} text={card.text} type={card.type}/>
-                  )}
-                </div>
-              </li>
+              {offers.map((offer) => {
+                const {id, type, isPremium, previewImage, price, rating, title, city, isFavorite} = offer;
+                return cityMain.map((location) => {
+                  if (location.city === city.name && isFavorite === true) {
+                    return (
+                      <li key={offer.id} className="favorites__locations-items">
+                        <div className="favorites__locations locations locations--current">
+                          <div className="locations__item">
+                            <a className="locations__item-link" href="#">
+                              <span>{location.city}</span>
+                            </a>
+                          </div>
+                        </div>
+                        <div className="favorites__places">
+                          <Card
+                            key={offer.id}
+                            id={id}
+                            typeCard={TypeCard.Favorites}
+                            isPremium={isPremium}
+                            previewImage={previewImage}
+                            price={price}
+                            rating={rating}
+                            title={title}
+                            type={type}
+                          />
```

Comment:
Так будет сильно проще читаться. А достать нужные свойства из offer уже можно внутри Card

```suggestion
                          <Card
                            offer={offer}
                            typeCard={TypeCard.Favorites}
                          />
```

И второе, не нужно на сам <Card> вешать атрибут `key`, он у тебя находится на `<li>`